### ISSUE TITLE
Preprocerror errors in mainline campaigns cause UMC campaigns to cease appearing in campaign dialog (GNA #23228)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1447]
**[Original submission](https://gna.org/bugs/index.php?23228)** by vultraz on 2015-01-30

If a mainline campaign has a preprocessor error, and the user is returned to the main menu and then goes to Campaigns, only mainline campaigns will be listed, but no UMC ones.

This persists even after refreshing the cache with F5 from the main menu or fixing the offending WML and entering a game. The only way I've found to make UMC appear again is to quit and restart the game.

(Reproduced on Windows 8.1)
Release: 1.13.x
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by nemaara** [comment]
@Vultraz seems like the behavior for this issue is now that the game will exit, not return to the main menu screen (also we really probably wouldn't ship a release with preprocessor errors in it anyway). Can this issue be closed?

**Comment by ProditorMagnus** [comment]
Might be related to #1924

**Comment by CelticMinstrel** [comment]
I'm pretty sure F5 is supposed to bypass the cache though?

Anyway, this is a problem that only affects developers, so the priority is super-low. It could be improved, to be sure, but it's not like a Wesnoth release will be shipped with preprocessor errors in the mainline campaigns.

**Comment by Wedge009** [comment]
On that note, I was thinking how to replicate this, since I don't see preprocessor errors in the main campaigns.

**Comment by gfgtdf** [comment]
> On that note, I was thinking how to replicate this, since I don't see preprocessor errors in the main campaigns.

Well you have  to add some to the main data file.

**Comment by GregoryLundberg** [comment]
Is it just main campaigns?

Can this be reproduced by introducing a preprocessing error into a UMC campaign with similar results?

If not, I don't see any reason to keep this issue open if ensuring there are no preprocessing issues in the mainline campaigns fully solves (avoids) the problem

**Comment by jostephd** [comment]
> Is it just main campaigns?

Yes. An error in an addon campaign doesn't prevent subsequent add-on campaigns from being loaded. It results in [a click-through dialog](https://github.com/wesnoth/wesnoth/blob/edf5765e7ce0209e0965ec1c70c4cf69b22a53cc/src/game_config_manager.cpp#L481-L494) saying the add-on has a bug.

The failure mode for a preprocessor error in a mainline campaign (an unmatched `#ifdef` in HttT/_main.cfg) is [two click-through messages](https://github.com/wesnoth/wesnoth/blob/edf5765e7ce0209e0965ec1c70c4cf69b22a53cc/src/game_config_manager.cpp#L308-L333) and then wesnoth exits.

> If not, I don't see any reason to keep this issue open if ensuring there are no preprocessing issues in the mainline campaigns fully solves (avoids) the problem

Well, maybe. It's low priority, certainly, but a WML error in a campaign shouldn't render the whole game unusable. It would be better to continue with that one campaign disabled, just as it's done for add-ons.
